components/brave_rewards/browser/database/publisher_info_database.cc

```diff
@@ -1607,11 +1702,85 @@ bool PublisherInfoDatabase::Migrate(int version) {
     case 7: {
       return MigrateV6toV7();
     }
+    case 8: {
+      return MigrateV7toV8();
+    }
     default:
       return false;
   }
 }
 
+bool PublisherInfoDatabase::MigrateDBTable(
+    const std::string& from,
+    const std::string& to,
+    const std::vector<std::string>& columns,
+    const bool should_drop) {
+  std::map<std::string, std::string> new_columns;
+  for (const auto& column : columns) {
+    new_columns[column] = column;
+  }
+
+  return MigrateDBTable(from, to, new_columns, should_drop);
+}
+
+bool PublisherInfoDatabase::MigrateDBTable(
+    const std::string& from,
+    const std::string& to,
+    const std::map<std::string, std::string>& columns,
+    const bool should_drop) {
+  DCHECK_NE(from, to);
+  DCHECK(!from.empty());
+  DCHECK(!to.empty());
+
+  std::string sql = "PRAGMA foreign_keys = off;";
+
+  if (!columns.empty()) {
+    const auto insert = CreateDBInsertQuery(from, to, columns);
+    sql.append(insert);
+  }
+
+  if (should_drop) {
+    sql.append(base::StringPrintf("DROP TABLE %s;", from.c_str()));
+  }
+
+  sql.append("PRAGMA foreign_keys = on;");
+
+  return GetDB().Execute(sql.c_str());
+}
+
+bool PublisherInfoDatabase::RenameDBTable(
+    const std::string& from,
+    const std::string& to) {
+  DCHECK_NE(from, to);
+
+  const auto sql = base::StringPrintf("ALTER TABLE %s RENAME TO %s;",
+      from.c_str(), to.c_str());
+
+  return GetDB().Execute(sql.c_str());
+}
+
+std::string PublisherInfoDatabase::CreateDBInsertQuery(
+    const std::string& from,
+    const std::string& to,
+    const std::map<std::string, std::string>& columns) {
+  DCHECK_GT(columns.size(), 0UL);
+
+  std::vector<std::string> from_columns;
+  std::vector<std::string> to_columns;
+
+  for (const auto& column : columns) {
+    from_columns.push_back(column.first);
+    to_columns.push_back(column.second);
+  }
+
+  const auto comma_separated_from_columns = base::JoinString(from_columns, ",");
+  const auto comma_separated_to_columns = base::JoinString(to_columns, ",");
+
+  return base::StringPrintf("INSERT INTO %s (%s) SELECT %s FROM %s;",
```

Comment:
nit: I would place every param in separate line as it's easier to read for me. Like this
```
return base::StringPrintf("INSERT INTO %s (%s) SELECT %s FROM %s;",
    to.c_str(),
    comma_separated_to_columns.c_str(),
    comma_separated_from_columns.c_str(),
    from.c_str());
```